cpp/modmesh/buffer/pymod/TypeBroadcast.hpp

```diff
@@ -71,8 +72,17 @@ struct TypeBroadcastImpl
                 offset_out += arr_out.stride(it) * sidx[it] * step;
             }
 
-            // NOLINTNEXTLINE(bugprone-signed-char-misuse, cert-str34-c)
-            arr_out.at(offset_out) = static_cast<out_type>(*ptr_in);
+            constexpr bool valid_conversion = (!is_complex_v<T> && !is_complex_v<D>) || (is_complex_v<T> && is_complex_v<D> && std::is_same_v<T, D>);
+
+            if constexpr (valid_conversion)
+            {
+                arr_out.at(offset_out) = static_cast<out_type>(*ptr_in);
+            }
+            else
+            {
+                throw std::runtime_error("Cannot convert between complex and non-complex types");
+            }
```

Comment:
Currently, only support `float` complex to `float` complex and `double` complex to `double` complex conversion.